.github/workflows/ci.yml

```diff
@@ -30,6 +30,29 @@ env:
 
 jobs:
 
+  # download & cache
+  #############################################################################
+
+  download_test_data:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/cache@v4
+        id: cache
+        with:
+          key: raw_test_data # fixed key will always hit; clear cache to trigger cache miss
+          path: clas_005038.evio.00000
+          lookup-only: true
+      - name: install xrootd-client
+        if: ${{ steps.cache.outputs.cache-hit != 'true' }}
+        run: |
+          sudo apt -y update
+          sudo apt -y upgrade
```

Comment:
`upgrade` sometimes gets stuck doing unrelated stuff; might as well avoid it, even if this step only ever runs one time (since fixed cache key always hits)
```suggestion
```